syntax = "proto3";

package pb;

service Analyzer {
  rpc analyze (AnalyzerInput) returns (AnalyzerOutput) {}
}

message AnalyzerInput {
  string id = 1;
  int64 timestamp = 2;
  string text = 3;
}

message AnalyzerOutput {
  message Category {
    string name = 1;
    double score = 2;
    double normed_score = 3;
  }

  repeated Category categories = 1;
}